File: Java/17-github-copilot-cli/src/main/java/com/githubcopilot/demo/TicketQueue.java
package com.githubcopilot.demo;

import java.time.Instant;
import java.util.List;
import java.util.Objects;

public final class TicketQueue {

    public enum TicketPriority {
        LOW,
        NORMAL,
        HIGH,
        CRITICAL
    }

    public record SupportTicket(
            String id,
            String title,
            TicketPriority priority,
            Instant createdAt,
            String owner) {
    }

    public record OwnerWorkload(String owner, int ticketCount) {
    }

    private final List<SupportTicket> tickets;

    public TicketQueue(List<SupportTicket> tickets) {
        Objects.requireNonNull(tickets, "tickets cannot be null");
        this.tickets = List.copyOf(tickets);
    }

    public List<SupportTicket> orderForTriage() {
        // TODO (Copilot CLI): implement the behavior specified in TASK.md.
        throw new UnsupportedOperationException("Not implemented");
    }

    public List<OwnerWorkload> getOwnerWorkload() {
        // TODO (Copilot CLI): implement the behavior specified in TASK.md.
        throw new UnsupportedOperationException("Not implemented");
    }
}

File: CSharp/17-github-copilot-cli/src/IssueDesk/TicketQueue.cs
namespace IssueDesk;

public enum TicketPriority
{
    Low,
    Normal,
    High,
    Critical
}

public sealed record SupportTicket(
    string Id,
    string Title,
    TicketPriority Priority,
    DateTimeOffset CreatedAt,
    string? Owner);

public sealed record OwnerWorkload(string Owner, int TicketCount);

public sealed class TicketQueue
{
    private readonly IReadOnlyList<SupportTicket> _tickets;

    public TicketQueue(IEnumerable<SupportTicket> tickets)
    {
        ArgumentNullException.ThrowIfNull(tickets);
        _tickets = tickets.ToList();
    }

    public IReadOnlyList<SupportTicket> OrderForTriage()
    {
        // TODO (Copilot CLI): implement the behavior specified in TASK.md.
        throw new NotImplementedException();
    }

    public IReadOnlyList<OwnerWorkload> GetOwnerWorkload()
    {
        // TODO (Copilot CLI): implement the behavior specified in TASK.md.
        throw new NotImplementedException();
    }
}
